code/game/machinery/droneDispenser.dm

```diff
@@ -126,114 +155,158 @@
 
 /obj/machinery/droneDispenser/examine(mob/user)
 	..()
-	if(droneMadeRecently && !stat && recharging_text)
+	if((mode == DRONE_RECHARGING) && !stat && recharging_text)
 		user << "<span class='warning'>[recharging_text]</span>"
 	if(stat & BROKEN)
-		user << "<span class='warning'>[src] is smoking and steadily buzzing. It seems to be broken.</span>"
+		user << "<span class='warning'>[src] is smoking and steadily buzzing. \
+			It seems to be broken.</span>"
 	if(metal_cost)
-		user << "<span class='notice'>It has [materials.amount(MAT_METAL)] units of metal stored.</span>"
+		user << "<span class='notice'>It has [materials.amount(MAT_METAL)] \
+			units of metal stored.</span>"
 	if(glass_cost)
-		user << "<span class='notice'>It has [materials.amount(MAT_GLASS)] units of glass stored.</span>"
+		user << "<span class='notice'>It has [materials.amount(MAT_GLASS)] \
+			units of glass stored.</span>"
 
 /obj/machinery/droneDispenser/power_change()
 	..()
-	if(stat & BROKEN)
-		return
+	if(powered())
+		stat &= ~NOPOWER
 	else
-		if(powered())
-			stat &= ~NOPOWER
-		else
-			icon_state = "off"
-			stat |= NOPOWER
+		stat |= NOPOWER
+	update_icon()
 
 /obj/machinery/droneDispenser/process()
 	..()
-	if(stat & (NOPOWER|BROKEN) || !anchored)
+	if((stat & (NOPOWER|BROKEN)) || !anchored)
 		return
-	if((metal_cost && glass_cost) && materials.can_use_amount(mats=using_materials))
+
+	if(!materials.has_materials(using_materials))
+		return // We require more minerals
+
+	// We are currently in the middle of something
+	if(timer > world.time)
 		return
-	if(droneMadeRecently)
+
+	switch(mode)
+		if(DRONE_READY)
+			// If we have X drone shells already on our turf
```

Comment:
actually, someone else while reviewing was confused about the various function of some of these calls, so I put in more comments.